src/App.tsx

```diff
@@ -1,20 +1,28 @@
 import { useEffect, useState } from "react";
 import "./App.css";
-import { useOddMode } from "./queries";
+import { useEnterOddMode, useOddMode } from "./queries";
 import { OddModeAnimation } from "./OddModeAnimation";
 import { log } from "./utils";
 
 function App() {
   const { data, isLoading } = useOddMode();
   // Demonstrate a modal dialog that's in the DOM
   const [showModal, setShowModal] = useState(false);
+  const enterMutation = useEnterOddMode();
 
   useEffect(() => {
-    if (data?.changed && !data.oddMode) {
-      log("show modal");
-      setShowModal(true);
+    if (data?.changed) {
+      if (!data.oddMode) {
+        log("show modal");
+        setShowModal(true);
```

Comment:
For some reason this change also breaks the ok button making this code run until the API called again.